images/macos/provision/configuration/CreateCleanAnkaTemplate.ps1

```diff
@@ -0,0 +1,232 @@
+    }
+
+    Write-Host "All the attempts exhausted, try again later"
+    exit 1
+}
+
+function Test-SoftwareUpdate {
+    param (
+        [String] $UpdateProcessName = "softwareupdate",
+        [Int] $RetryCount = 20,
+        [Int] $RetryInterval = 60
+    )
+
+    while ($RetryCount -ne 0)
+    {
+        $RetryCount--
+        $updateProcess = (Get-Process -Name $UpdateProcessName -ErrorAction SilentlyContinue).id
+
+        if (-not $updateProcess)
+        {
+            return
+        }
+
+        # Workaround to get commandline param as it doesn't work for macOS atm https://github.com/PowerShell/PowerShell/issues/13943
+        $processName = ps -o command= $updateProcess
+        Write-Host "Another software update process is in place with the following arguments '$processName', wait $RetryInterval seconds, $RetryCount attempts left"
+        Start-Sleep -Seconds $RetryInterval
+    }
+
+    Write-Host "All the attempts exhausted, try again later"
+    exit 1
+}
+
+function New-AnkaVMTemplate {
+    param (
+        [String] $InstallerPath,
+        [String] $ShortMacOSVersion,
+        [String] $TemplateName,
+        [String] $TemplateUsername,
+        [String] $TemplatePassword,
+        [Int] $CpuCount,
+        [Int] $RamSizeGb,
+        [Int] $DiskSizeGb,
+        [String] $DisplayResolution
+    )
+
+    $isTemplateExists = (Invoke-Anka { anka --machine-readable list } | ConvertFrom-Json).body.name -eq $templateName
+    if ($isTemplateExists)
+    {
+        Write-Host "Deleting existed template with name '$templateName' before creating a new one"
+        Invoke-Anka { anka delete $templateName --yes }
+    }
+
+    Write-Host "Creating Anka VM template with name '$templateName' and user $TemplateUsername"
+    $env:ANKA_DEFAULT_USER = $TemplateUsername
+    $env:ANKA_DEFAULT_PASSWD = $TemplatePassword
+    $env:ANKA_CREATE_SUSPEND = 0
+    Invoke-Anka { anka create --cpu-count $CpuCount --ram-size "${RamSizeGb}G" --disk-size "${DiskSizeGb}G" --app $InstallerPath $templateName }
+
+    # Apple Metal is available starting from Big Sur
+    if (-not $ShortMacOSVersion.StartsWith("10"))
+    {
+        Write-Host "Enabling Graphics Acceleration with Apple Metal for $templateName"
+        Invoke-Anka { anka modify $templateName set display -c pg }
+    }
+    Write-Host "Setting screen resolution to $DisplayResolution for $templateName"
+    Invoke-Anka { anka modify $templateName set display -r $DisplayResolution }
+
+    return $templateName
+}
+
+function Add-AnkaImageToRegistry {
+    param (
+        [String] $RegistryUrl,
+        [String] $ShortMacOSVersion,
+        [String] $TemplateName
+    )
+
+    Write-Host "Adding Anka remote registry to the list of registries"
+    Invoke-Anka { anka registry add ankaregistry $RegistryUrl  }
+
+    Write-Host "Setting up default registry for push"
+    Invoke-Anka { anka registry set ankaregistry }
+
+    Write-Host "Pushing clean macOS $ShortMacOSVersion to the $RegistryUrl"
```

Comment:
Let's use -a and no need to add /set registry, just `Invoke-Anka { anka registry -a $RegistryUrl push -t $ShortMacOSVersion $TemplateName }`